abel/tools/circularize.py

```diff
@@ -0,0 +1,236 @@
+# -*- coding: utf-8 -*-
+from __future__ import division
+import numpy as np
+from scipy import ndimage
+from scipy.interpolate import UnivariateSpline
+from scipy.optimize import leastsq
+
+import abel
+
+# see https://github.com/PyAbel/PyAbel/issues/186 for discussion
+
+
+def circularize_image(IM, method="lsq", center=None, radial_range=None,
+                      zoom=1, smooth=0, nslices=32, ref_angle=None, 
+                      inverse=False, return_correction=False):
+    """
+    Remove radial distortion from a velocity-map-image through radial scaling
+    of the Newton-rings to enforce circularity.
+
+    The image is divided into angular slices, adjacent radial intensity
+    profile slices compared to give a radial scaling factor that best
+    aligns common structure.
+
+    The resultant radial scaling factor vs angle is interpolated to apply the
+    correction to the whole image grid. The image is remapped to the
+    correct cartesian grid.
+
+    Parameters
+    ----------
+    IM : numpy 2D array
+
+    method: str
+        method to align slice profiles,
+        "argmax" - compare intensity-profile.argmax() of each radial slice.
+        "lsq" - determine radial scaling factor by minimizing the
+                intensity-prfoile with an adjacent slice intensity-profile.
+
+    center: str, or None
+        pre-center image using PyAbel centering methods,
+        "com", "convolution", "gaussian", "image_center", "slice"
+
+    radial_range: tuple, or None
+        limit slice comparison to the radial range (rmin, rmax).
+
+    zoom: float
+        nmimage.zoom image before analysis. This may help improve the low
+        radius slice profile.
+
+    smooth: float
+        smoothing for spline interpolation of the determined radial scaling
+        factor vs angle
+
+    nslices: int
+        divide the image into nslices. Ideally, this should be a divisor of
+        the image column width.
+
+    ref_angle: None or float
+        reference angle for which radial coordinate is unchanged.
+        None uses numpy.mean(radial scale factors)
+
+    inverse: bool
+        inverse Abel transform the *polar* image, to remove the background
+        intensity. This helps follow a given peak intensityr, for all angles,
```

Comment:
I'm not sure that this would *only* help when the anisotropy parameter is large. It seems like it might improve the SNR in many situations. 

Also, worth mentioning: "Note that this step is only for the purposes of allowing the algorithm to better follow peaks in the image. It does not affect the final image that is returned, expect for (hopefully) slightly improving the precision of the distortion correction." 

"intensityr" -> "intensity"